gdal/port/cpl_vsil.cpp

```diff
@@ -2362,79 +2361,52 @@ static VSIFileManager *poManager = nullptr;
 static CPLMutex* hVSIFileManagerMutex = nullptr;
 
 VSIFileManager *VSIFileManager::Get()
-
 {
-    static volatile GPtrDiff_t nConstructerPID = 0;
-    if( poManager != nullptr )
-    {
-        if( nConstructerPID != 0 )
-        {
-            GPtrDiff_t nCurrentPID = static_cast<GPtrDiff_t>(CPLGetPID());
-            if( nConstructerPID != nCurrentPID )
-            {
-                {
-                    CPLMutexHolder oHolder( &hVSIFileManagerMutex );
-                }
-                // This construct to avoid a lock is highly questionable
-                // and could probably fail with out-of-order CPU execution
-                // so the cppcheck warning is probably quite relevant.
-                // cppcheck-suppress identicalInnerCondition
-                if( nConstructerPID != 0 )
-                {
-                    VSIDebug1( "nConstructerPID != 0: %d", nConstructerPID);
-                    assert(false);
-                }
-            }
-        }
+      CPLMutexHolder oHolder(&hVSIFileManagerMutex);
+      if ( poManager != nullptr ) {
         return poManager;
-    }
+      }
 
-    CPLMutexHolder oHolder2( &hVSIFileManagerMutex );
-    if( poManager == nullptr )
-    {
-        nConstructerPID = static_cast<GPtrDiff_t>(CPLGetPID());
 #ifdef DEBUG_VERBOSE
-        printf("Thread " CPL_FRMT_GIB": VSIFileManager in construction\n",  // ok
-               static_cast<GIntBig>(nConstructerPID));
+      printf("Thread " CPL_FRMT_GIB ": VSIFileManager in construction\n" /*ok*/);
```

Comment:
CPL_FRMT_GIB requires an argument. But you can just remove the #ifdef DEBUG_VERBOSE / #endif completely, here and and at the end of the function

Reply:
Done.